### DISCUSSION TITLE
Renovate requires GitHub token when adding RENOVATE_IGNORE_GIT_AUTHOR env variable

### DISCUSSION THREAD

**Discussion by p3k** [renovatebot/renovate#32253] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Forgejo at Codeberg.org, latest Renovate

### Please tell us more about your question or problem

I have a Forgejo workflow that works fine but as soon as I add the `RENOVATE_IGNORE_GIT_AUTHOR` environment variable Renovate begins to require a GitHub token:

```plain
FATAL: You must configure a GitHub token
```

I made a [repro at Codeberg.org](https://codeberg.org/repros/renovate-git-ignored-authors/src/branch/main/.github/workflows/renovate.yml):

* [Worfklow](https://codeberg.org/repros/renovate-git-ignored-authors/actions/runs/10) runs Renovate two times
* First time without `RENOVATE_IGNORE_GIT_AUTHOR` – succeeds
* Second time with – fails

### Logs (if relevant)

* [Log of a successful run](https://codeberg.org/repros/renovate-git-ignored-authors/actions/runs/10/artifacts/renovate-log-without.ndjson)
* Log of a failed run is not available (presumably Renovate errors before starting to write it?)

**Comment by rarkins** [comment]
Where do I find this FATAL error in your logs?

**Reply by p3k** [reply]
Ah! I think they are the same file. My bad, it looks like in the second run Renovate does not even begin to write a log file – and then the existing one from the first run is uploaded as artifact again, just with a different name. 

You can see the FATAL error in the workflow output:

![codeberg org_repros_renovate-git-ignored-authors_actions_runs_9](https://github.com/user-attachments/assets/bbf4cec8-1f63-4678-8005-5696924c6d0b)

**Reply by p3k** [reply]
I modified the [workflow](https://codeberg.org/repros/renovate-git-ignored-authors/commit/a58e9f0762386edcabd2cbed964e7132520646ca) so each Renovate run writes to its own log file (see [run #10](https://codeberg.org/repros/renovate-git-ignored-authors/actions/runs/10)) – and indeed there is no file to upload from the second run:

![codeberg org_repros_renovate-git-ignored-authors_actions_runs_10](https://github.com/user-attachments/assets/b9b2b6ec-7aee-4b35-8aa9-3e2cf4f3c0a1)

**Reply by p3k** [reply]
When I [increase `LOG_LEVEL` to `debug`](https://codeberg.org/repros/renovate-git-ignored-authors/commit/6997fa796ff6c1c2270d8a85cf392bd00212833a) Renovate still does not produce more output, and even still claims the log level was set to `info` – [run #11](https://codeberg.org/repros/renovate-git-ignored-authors/actions/runs/11) :shrug: 

> INFO: Renovate was run at log level "info". Set LOG_LEVEL=debug in environment variables to see extended debug logs.

**Reply by rarkins** [reply]
If you see that INFO message, it means you have not set LOG_LEVEL=debug in a way that works

**Reply by p3k** [reply]
Well, if I set `LOG_LEVEL: debug` in the `env` section of the succeeding job I see debug messages – [run #13](https://codeberg.org/repros/renovate-git-ignored-authors/actions/runs/13).

If I invoke Renovate with `LOG_LEVEL=debug npx renovate` I see debug messages in the failing job, too: it looks like [Renovate does not find the configuration file](https://codeberg.org/repros/renovate-git-ignored-authors/actions/runs/13#jobstep-3-6)…? Are all the `env` settings ignored as soon as I add `RENOVATE_IGNORE_GIT_AUTHOR` there? :thinking:

**Reply by rarkins** [reply, marked as answer]
No, they're not ignored. The problem sounds like it's with your environment still

**Reply by p3k** [reply]
Yeah, I phrased that wrong before: to me it looks like [the `env` settings in the failing step are not applied to the Renovate configuration](https://codeberg.org/repros/renovate-git-ignored-authors/actions/runs/13#jobstep-3-11) compared to as it is [logged in the succeeding step](https://codeberg.org/repros/renovate-git-ignored-authors/actions/runs/13#jobstep-1-13):

```plain
DEBUG: Env config
       "config": {"hostRules": []}

vs

DEBUG: Env config
       "config": {
         "hostRules": [
           {"hostType": "github", "matchHost": "github.com", "token": "***********"}
         ],
         "repositoryCache": "enabled",
         "logFile": "renovate-log-without.ndjson",
         "logFileLevel": "debug",
         "platform": "gitea",
         "endpoint": "https://codeberg.org/api/v1",
         "token": "***********",
         "autodiscover": true,
         "gitAuthor": "Renovate Bot <bot@renovateapp.com>"
       }
```

@rarkins So are you implying it is rather an issue with the CI/CD code of Codeberg/Forgejo than with Renovate?

**Reply by rarkins** [reply]
Yes

**Reply by p3k** [reply]
OK, makes sense. Gonna create an issue there, too. Thanks for the feedback! :)
